lib/astutils.cpp

```diff
@@ -1023,6 +1023,73 @@ static bool isZeroBoundCond(const Token * const cond)
     return false;
 }
 
+static bool isForLoopCondition(const Token * const tok)
+{
+    if (!tok || !tok->astParent())
+        return false;
+    const Token *const parent = tok->astParent();
+    return parent->str() == ";" && parent->astOperand1() == tok &&
+           parent->astParent() && parent->astParent()->str() == ";" &&
+           parent->astParent()->astParent() && parent->astParent()->astParent()->str() == "(" &&
+           parent->astParent()->astParent()->astOperand1()->str() == "for";
+}
+
+/**
+ * Is token used a boolean (cast to a bool, or used as a condition somewhere)
+ * @param tok
+ * @param checkingParent true if we are checking a parent. This is used to know
+ * what we are checking. For instance in `if (i == 2)`, isUsedAsBool("==") is
+ * true whereas isUsedAsBool("i") is false, but it might call
+ * isUsedAsBool_internal("==") which must not return true
+ */
+static bool isUsedAsBool_internal(const Token * const tok, bool checkingParent)
+{
+    if (!tok)
+        return false;
+    const Token::Type type = tok->tokType();
+    if (type == Token::eBitOp || type == Token::eIncDecOp || (type == Token::eArithmeticalOp && (tok->str() != "*" || tok->astOperand2())))
```

Comment:
suggestion; replace `(tok->str() != "*" || tok->astOperand2())` with `!tok->isUnaryOp("*")`

Reply:
It seems that has been resolved!?